Src/Sarif.PatternMatcher.Sdk/Fingerprint.cs

```diff
@@ -106,12 +101,14 @@ private enum ParseState
 
         public string Secret { get; set; }
 
-        public string Resource { get; set; }
+        public bool IgnorePath { get; set; }
```

Comment:
>IgnorePath [](http://example.com/codeflow?start=20&length=10)

oh, you need this because this is a struct and it's required for all instantiation patterns?

Reply:
Not only that. if we add a field and you  use:

new fingerprint
{
 a = b
}

this won't trigger the constructor and, we can not have a constructor without initializing all parameters and without parameters